src/app/profile/edit-profile/edit-profile.component.ts

```diff
@@ -1,10 +1,68 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
+import { Profile } from '../../shared/models/profile.model';
+import { ProfileService } from './profile.service';
+import { Router } from '@angular/router';
+import { environment } from '../../../env/environment';
 
 @Component({
   selector: 'app-edit-profile',
   templateUrl: './edit-profile.component.html',
-  styleUrl: './edit-profile.component.css'
+  styleUrls: ['./edit-profile.component.css']
 })
-export class EditProfileComponent {
+export class EditProfileComponent implements OnInit {
+  profile: Profile | null;
 
+  constructor(
+    private profileService: ProfileService,
+    private router: Router
+  ) { }
+
+  ngOnInit(): void {
+    this.profileService.getProfile().subscribe({
+      next: (data: Profile | null) => {
+        this.profile = data;
+        console.log(this.profile);
+      }
+    });
+  }
+
+  onSaveClick(): void {
+    if (this.profile) {
+      this.profileService.updateProfile(this.profile).subscribe({
+        next: (_) => {
+          console.log("konzolugujem");
+        }
+      }
+      );
+    } else {
+      console.error('Profile not found');
+    }
```

Comment:
Don't send a PUT request if some of the fields are not valid. 
```suggestion
    } else if (!this.valid) {
      this.sharedService.openSnack('Please fill in all required fields');
    } 
    else {
      console.error('Profile not found');
    }
```